### ISSUE TITLE
Null Wearable is added to inventory

### ISSUE THREAD

**Issue by mikhail-dcl** [decentraland/unity-renderer#4167]
https://play.decentraland.zone/?BUILDER_SERVER_URL=https://builder-api.decentraland.zone/v1&NETWORK=goerli&DEBUG_MODE=true&WITH_COLLECTIONS=9c4d923f-3e19-4abc-a012-33397f1e10ea

You will receive a malformed WearablesRequestFailed: `(intermediate value).data.filter is not a function`
The correct structure is:
```
[Serializable]
public class WearablesRequestFailed
{
    public string error;
    public string context;
}
```
So it's impossible to execute:
```
if (requestFailedResponse.context == BASE_WEARABLES_CONTEXT ||
            requestFailedResponse.context.Contains(THIRD_PARTY_WEARABLES_CONTEXT) ||
            requestFailedResponse.context.Contains(OWNED_WEARABLES_CONTEXT))
        {
            ResolvePendingWearablesByContextPromise(
                requestFailedResponse.context,
                null,
                requestFailedResponse.error);
        }
        else
        {
            string[] failedWearablesIds = requestFailedResponse.context.Split(',');
            for (int i = 0; i < failedWearablesIds.Length; i++)
            {
                ResolvePendingWearablePromise(
                    failedWearablesIds[i],
                    null,
                    $"The request for the wearable '{failedWearablesIds[i]}' has failed: {requestFailedResponse.error}");
            }
        }
```
We can prevent an NRE by
```
if (requestFailedResponse?.context == null)
            return;
```

Then you will see:
```
PromiseException: The request for the wearable 'urn:decentraland:off-chain:base-avatars:Espadrilles' has failed: (intermediate value).data.filter is not a function
```
It looks like after that `null` is added to the inventory and causes many other `Null Reference Exception`s and it becomes a blocker.

We should find a place to prevent adding `null` to the inventory

**Comment by sandrade-dcl** [comment]
I'm trying to reproduce the issue running: https://play.decentraland.zone/?BUILDER_SERVER_URL=https://builder-api.decentraland.zone/v1&NETWORK=goerli&DEBUG_MODE=true&WITH_COLLECTIONS=9c4d923f-3e19-4abc-a012-33397f1e10ea

Although I'm able to reproduce the NullRef exception on our side (in `CatalogController.WearablesRequestFailed` method), I'm seeing that exception is not breaking the user's flow neither adding a "null" wearable into the catalog.

I have asked Lautaro Petaccio for some help to try to set my userId into the same scenario as the user who reported the issue.

**Comment by sandrade-dcl** [comment]
Ok, I have detected a wrong behavior of the backpack which caused the it to stuck when the request of the owned wearables failed:

![image.png](https://images.zenhubusercontent.com/5eb2b8ca65c88f6e391ef16e/79f6ab57-fb5b-4f3d-9641-c523e7129120)

Maybe it is related to what happened in the original issue (https://decentralandteam.slack.com/archives/C01QVR7TJK1/p1675080444432059). I have created a PR with 2 fixes:

- A NullRef that we were not managing correctly in the `CatalogController.WearablesRequestFailed` method.
- A wrong behavior of the backpack which caused it to stuck when the request of the owned wearables failed.
